chain/rosetta-rpc/src/adapters/mod.rs

```diff
@@ -136,38 +133,41 @@ pub(crate) async fn convert_block_to_transactions(
                 None
             }
         })
-        .filter(move |account_id| {
-            // TODO(mina86): Convert this to seen.get_or_insert_with(account_id,
-            // Clone::clone) once hash_set_entry stabilises.
-            seen.insert(account_id.clone())
-        })
-        .collect::<Vec<_>>();
+        .collect::<std::collections::HashSet<_>>();
```

Comment:
Using a hash set instead of a vector changes the order of the values in the final `Vec<_>` returned, right?

I'm not sure if this is important. But the comment on `fn block_details()` says something about being idempotent and requiring the exact same block contents. I wonder if equivalent block content, represented in a different order is also acceptable.

I think that's for you and @firatNEAR to decide, as I have no stake in the Rosetta RPC API and the guarantees it makes.